### PR TITLE
add ability to restore cluster from snapshots #96

### LINKED ISSUE #96: Support restoring from a snapshot
Have a question? Please checkout our [Slack Community](https://slack.cloudposse.com) or visit our [Slack Archive](https://archive.sweetops.com/). 

[Slack Community](https://slack.cloudposse.com)

## Describe the Feature

The elasticache replication group resource allows an optional parameter to restore from a snapshot. This module should support this parameter.

## Expected Behavior

When snapshot_arns passed into the module, the created cluster is created from the snapshot. By default this should just be an empty list.

## Use Case

From recovering from an older version of the redis cluster

## Describe Ideal Solution

New variable snapshot_arns added and passed to the replication group resource

## Alternatives Considered

Explain what alternative solutions or features you've considered.

### PR DESCRIPTION
**Pull request by skang0601**
## what
Allows the ability to create redis cluster from snapshots

## references
* closes #96

### CODE REVIEW

**Review by kurtmc — APPROVED** [review]
This looks good to me. Is there anything stopping this from being merged?

**Comment by skang0601** [comment]
@kurtmc Merge conflict has been addressed. I don't think I have permissions to merge to this repo, do these changes look good for y'all?

File: variables.tf [hunk 1]
```diff
@@ -175,6 +175,12 @@ variable "replication_group_id" {
   default     = ""
 }
 
+variable "snapshot_arns" {
+  type        = list(string)
+  description = "(Optional) A single-element string list containing an Amazon Resource Name (ARN) of a Redis RDB snapshot file stored in Amazon S3. Example: arn:aws:s3:::my_bucket/snapshot1.rdb"
```
Comment (SweetOps) [inline variables.tf:180]:
```suggestion
  description = "A single-element string list containing an Amazon Resource Name (ARN) of a Redis RDB snapshot file stored in Amazon S3. Example: arn:aws:s3:::my_bucket/snapshot1.rdb"
```

for consistency with other variables

**Review by SweetOps — CHANGES_REQUESTED** [review]

**Review by SweetOps — APPROVED** [review]

**Pull request merged**